### PR TITLE
Add presubmit jobs for Cluster API DO provider

### PR DESCRIPTION
**Pull request by nikhita**
This PR

* Converts Circle CI tests to Prowjobs and adds presubmit jobs for build, test, verify and lint.
* Adds the relevant entries for testgrid.
* Enables the trigger plugin for the repo.

### CODE REVIEW

**Comment by nikhita** [comment]
/hold 
until https://github.com/kubernetes-sigs/cluster-api-provider-digitalocean/pull/118 is merged.

File: testgrid/cmd/configurator/config_test.go [hunk 1]
```diff
@@ -346,6 +346,7 @@ func TestJobsTestgridEntryMatch(t *testing.T) {
 		"kubernetes-sigs/aws-ebs-csi-driver",
 		"kubernetes-sigs/cluster-api",
 		"kubernetes-sigs/cluster-api-provider-aws",
+		"kubernetes-sigs/cluster-api-provider-digital-ocean",
```
[the remark this answers is not in the document]
Comment (nikhita) [reply testgrid/cmd/configurator/config_test.go]: Fixed.

File: prow/config.yaml [hunk 2]
```diff
@@ -294,13 +294,7 @@ branch-protection:
         cluster-api-provider-digitalocean:
           required_status_checks:
             contexts:
-            - "ci/circleci: build"
-            - "ci/circleci: check-dependencies"
-            - "ci/circleci: checkout_code"
             - "ci/circleci: e2e"
```
[the remark this answers is not in the document]
Comment (xmudrii) [reply prow/config.yaml]: No. Actually that checks only spawns Minikube cluster and that's it. As we're switching to Prow, it's probably going to change how we spawn Minikube, so removing it should be okay.

**Comment by nikhita** [comment]
/hold cancel
https://github.com/kubernetes-sigs/cluster-api-provider-digitalocean/pull/118 has merged.

**Comment by nikhita** [comment]
@fejta I've updated this PR with the feedback you had for publishing-bot (https://github.com/kubernetes/test-infra/pull/9895). Can you take a look at this one as well? Thanks!

File: config/jobs/kubernetes-sigs/cluster-api-provider-digitalocean/cluster-api-provider-digitalocean-presubmits.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        - "--job=$(JOB_NAME)"
+        - "--repo=sigs.k8s.io/$(REPO_NAME)=$(PULL_REFS)"
+        - "--service-account=/etc/service-account/service-account.json"
+        - "--upload=gs://kubernetes-jenkins/pr-logs"
+        - "--scenario=kubernetes_bazel"
+        - "--timeout=15"
+        - "--" # end bootstrap args, scenario args below
+        - "--install=gubernator/test_requirements.txt"
+        - "--test=//..."
+        - "--test-args=--config=lint"
```
Comment (nikhita) [inline config/jobs/kubernetes-sigs/cluster-api-provider-digitalocean/cluster-api-provider-digitalocean-presubmits.yaml]: I'm not entirely sure how to use pod utilities for this. :thinking:

Comment (krzyzacy) [reply config/jobs/kubernetes-sigs/cluster-api-provider-digitalocean/cluster-api-provider-digitalocean-presubmits.yaml]: you can do something like

```
containers:
      - image: gcr.io/k8s-testimages/kubekins-e2e:v20181024-be2f242dd-experimental
        command:
        - runner.sh
        - kubernetes_bazel.py
        args:
        - "--install=gubernator/test_requirements.txt"
        - "--test=//..."
        - "--test-args=--config=lint"
```

Comment (nikhita) [reply config/jobs/kubernetes-sigs/cluster-api-provider-digitalocean/cluster-api-provider-digitalocean-presubmits.yaml]: @krzyzacy Thanks! I've updated it.

File: config/jobs/kubernetes-sigs/cluster-api-provider-digitalocean/cluster-api-provider-digitalocean-presubmits.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        command:
+        - make
+        args:
+        - verify
+  - name: pull-cluster-api-provider-digitalocean-lint
+    always_run: true
+    decorate: true
+    path_alias: sigs.k8s.io/cluster-api-provider-digitalocean
+    spec:
+      containers:
```
Comment (krzyzacy) [inline config/jobs/kubernetes-sigs/cluster-api-provider-digitalocean/cluster-api-provider-digitalocean-presubmits.yaml]: sorry, that was wrong indent in my comment :-)

Comment (nikhita) [reply config/jobs/kubernetes-sigs/cluster-api-provider-digitalocean/cluster-api-provider-digitalocean-presubmits.yaml]: @krzyzacy Fixed :)

**Comment by nikhita** [comment]
Bump. 

Is there anything else to be done? PTAL. :)

**Comment by krzyzacy** [comment]
oops, sorry it's not assigned to anyone and missed from my radar
/lgtm

**Pull request merged**